app/grandchallenge/reader_studies/models.py

```diff
@@ -723,9 +795,12 @@ def validate(*, creator, question, answer, images, is_ground_truth=False):
                 f"{type(answer)} found."
             )
 
+    def calculate_score(self, ground_truth):
+        self.score = self.question.calculate_score(self.answer, ground_truth)
+        self.save()
```

Comment:
This could cause an infinite update loop, right? I guess the signal is filtered on `post_add` so probably ok. 

I think saving here goes beyond the responsibilities of this function. Probably better to remove the save method and use filter(...).update(...) in the signal.

Reply:
I think it should not cause an infinite loop, but I agree that it would be cleaner to explicitly call save if you want the score to be saved.

Reply:
moved.